From c9d21a58bea2ecb2dc8876e09e471094d54a25e4 Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Mon, 13 Apr 2020 01:08:12 +0200
Subject: [PATCH 1/3] Revert "PR/93: iaeiaeiaeiae: Do as the comment says, and
 count as dynamically linked"

This reverts commit 24c9c086cd7c55b7b0a003a145b32466468e2608.

Fixes misdetection of shared libraries as statically linked as
reported in bug 717264.

Bug: https://bugs.gentoo.org/717264
Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
---
 src/readelf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/readelf.c b/src/readelf.c
index 40bcfab1..afec31d1 100644
--- a/src/readelf.c
+++ b/src/readelf.c
@@ -1638,6 +1638,7 @@ dophn_exec(struct magic_set *ms, int clazz, int swap, int fd, off_t off,
 		/* Things we can determine before we seek */
 		switch (xph_type) {
 		case PT_DYNAMIC:
+			linking_style = "dynamically";
 			doread = 1;
 			break;
 		case PT_NOTE:
@@ -1653,7 +1654,6 @@ dophn_exec(struct magic_set *ms, int clazz, int swap, int fd, off_t off,
 			}
 			/*FALLTHROUGH*/
 		case PT_INTERP:
-			linking_style = "dynamically";
 			doread = 1;
 			break;
 		default:
-- 
2.26.0

